### PR TITLE
Adding support for creating Atoms in expressions

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Enables the `..` autoscoping style for creating Atoms in expressions. 
- Add type checking to methods in columns.
- Auto wrap returns from method in expressions into a column as needed.
- Remove `Time_Period.Day` to remove confusion..

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by Akirathan — APPROVED** [review]
Java changes OK. Please, don't forget to add tests for the new syntax in expressions, I have not seen them in your PR yet.

**Review by marthasharkey — APPROVED** [review]

File: std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java [hunk 1]
```diff
@@ -224,6 +226,29 @@ public Value visitUnaryMinus(ExpressionParser.UnaryMinusContext ctx) {
...
+  public Value visitAtom(ExpressionParser.AtomContext ctx) {
+    var atomName = ctx.IDENTIFIER().getText();
+    return makeConstructor.apply(toCamelCase(atomName));
+  }
+
+  private static String toCamelCase(String name) {
+    var builder = new StringBuilder(name.length());
+    boolean convertNext = true;
+    for (var c : name.toCharArray()) {
+      if (convertNext) {
```
Comment (hubertp) [inline std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java:239]: `Foo_Bar` will be `fooBar` but
`Foo__Bar` will be `foobar`

**Review by hubertp — APPROVED** [review]

File: std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java [hunk 2]
```diff
@@ -224,6 +226,29 @@ public Value visitUnaryMinus(ExpressionParser.UnaryMinusContext ctx) {
...
+        builder.append(Character.toUpperCase(c));
+        convertNext = false;
```
Comment (jdunkerley) [inline std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java]:
```suggestion
        convertNext = c == '_';
```

File: std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java (hunk 1 above)
Comment (jdunkerley) [reply std-bits/table/src/main/java/org/enso/table/expressions/ExpressionVisitorImpl.java:239]: First letter is capitalized but yes there was a bug with multiple `_`.

Should now map `fOO__baR` to `Foo__Bar`

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**